praw/models/reddit/subreddit.py

```diff
@@ -2849,6 +2857,55 @@ def remove(self, redditor: Union[str, "praw.models.Redditor"]):
         self.subreddit._reddit.post(url, data=data)
 
 
+class ModNotes:
```

Comment:
What do you think about having two classes; one for creating them and another for accessing/deleting mod notes on a user[s] and importing from toolbox/snoonotes (I already have code written for deflating the toolbox notes blob).

For the creation, we could have a `ModNoteMixin` that adds a `create_note` or similar method to submission and comment objects. It would also enable adding the mixin to other classes if Reddit add supports for more things (like modmail messages).

Reply:
I definitely like the idea of a create_note function on submissions and comments. I'll take a look at the best way to do that, but I feel like ModNoteMixin should just be add a helper function that calls to these create functions rather than actually split the code up.

Reply:
Added a mixin for this. It kinda just assumes the class it's called on is gonna have subreddit, author and fullname fields, but I think that's the right approach.